bpfd/src/server/bpf.rs

```diff
@@ -165,8 +187,13 @@ impl<'a> BpfManager<'a> {
                 return Ok(());
             }
 
+            // Cache program length so programs goes out of scope and
+            // sort_extensions() can generate its own list.
+            let program_len = programs.len() as u8;
+            self.sort_extensions(&if_index);
```

Comment:
Why do we sort here? It seems like it's less efficient - we sort right before attach at the moment so we only iterate over the list once...
In this code we iter at least once more to set `current_position` :thinking: